packages/superset-ui-plugin-chart-table/src/Table.tsx

```diff
@@ -0,0 +1,276 @@
+    }
+    this.setState({
+      selectedCells: newSelectedCells,
+    });
+  };
+
+  static getDerivedStateFromProps(props: TableProps, state: TableState) {
+    const { filters } = props;
+    const { selectedCells } = state;
+    const newSelectedCells = new Set(Array.from(selectedCells));
+    Object.keys(filters).forEach(key => {
+      filters[key].forEach(value => {
+        newSelectedCells.add(`${key}#${value}`);
+      });
+    });
+    return {
+      ...state,
+      selectedCells: newSelectedCells,
+    };
+  }
+
+  render() {
+    const {
+      metrics: rawMetrics,
+      timeseriesLimitMetric,
+      orderDesc,
+      percentMetrics,
+      data,
+      alignPositiveNegative,
+      colorPositiveNegative,
+      columns,
+      tableTimestampFormat,
+      tableFilter,
+    } = this.props;
+    const { selectedCells } = this.state;
+    const metrics = (rawMetrics || [])
+      .map(m => (m as Metric).label || (m as string))
+      // Add percent metrics
+      .concat((percentMetrics || []).map(m => `%${m}`))
+      // Removing metrics (aggregates) that are strings
+      .filter(m => typeof data[0][m as string] === 'number');
+    const dataArray: {
+      [key: string]: any;
+    } = {};
+
+    const sortByKey =
+      timeseriesLimitMetric &&
+      ((timeseriesLimitMetric as Metric).label || (timeseriesLimitMetric as string));
+
+    metrics.forEach(metric => {
+      const arr = [];
+      for (let i = 0; i < data.length; i += 1) {
+        arr.push(data[i][metric]);
+      }
+
+      dataArray[metric] = arr;
+    });
+
+    const maxes: {
+      [key: string]: number;
+    } = {};
+    const mins: {
+      [key: string]: number;
+    } = {};
+
+    for (let i = 0; i < metrics.length; i += 1) {
+      if (alignPositiveNegative) {
+        maxes[metrics[i]] = Math.max(...dataArray[metrics[i]].map(Math.abs));
+      } else {
+        maxes[metrics[i]] = Math.max(...dataArray[metrics[i]]);
+        mins[metrics[i]] = Math.min(...dataArray[metrics[i]]);
+      }
+    }
+
+    // const tsFormatter = getTimeFormatter(tableTimestampFormat);
+    let formattedData = data.map(row => ({
+      data: row,
+    }));
+
+    if (sortByKey) {
+      formattedData = formattedData.sort((a, b) => {
+        const delta = a.data[sortByKey] - b.data[sortByKey];
+        if (orderDesc) {
+          return -delta;
+        }
+        return delta;
+      });
+      if (metrics.indexOf(sortByKey) < 0) {
+        formattedData = formattedData.map(row => {
+          const data = { ...row.data };
+          delete data[sortByKey];
+          return {
+            data,
+          };
+        });
+      }
+    }
+
+    const tsFormatter = getTimeFormatter(tableTimestampFormat);
+
+    const heightType = 'small';
+    const getRenderer = (column: ColumnType) => (props: RendererProps) => {
+      const { key } = props;
+      let value = props.row.rowData.data[key];
+      const cellHash = `${key}#${value}`;
```

Comment:
Could define a method to get this to make sure it's the same here and in lifecycle methods